pynumaflow/function/server.py

```diff
@@ -242,6 +245,8 @@ async def __async_reduce_handler(self, interval_window, request_iterator: AsyncI
                 task = asyncio.create_task(
                     self.__invoke_reduce(key, riter, Metadata(interval_window=interval_window))
                 )
+                self.background_tasks.add(task)
```

Comment:
try without doing this and see whether we endup in weakref.. if yes, we will have to call this out in our doc.

Reply:
the second save is concerning as users unfamiliar with asyncio are bound to miss.

Reply:
The observation from the testing indicates that just using the second save works for us, but using only the first save continues to throw the error. 
Digging in to confirm the expectations for weak references in python.

Reply:
The references are not hierarchical in nature for the coroutines, child is not automatically referred by the parent.